Rollbar/DTOs/Request.cs

```diff
@@ -14,17 +28,104 @@ public class Request
         /// Initializes a new instance of the <see cref="Request"/> class.
         /// </summary>
         public Request()
-            : this(null)
+            : base(null)
         {
         }
 
+#if (NETSTANDARD || NETCOREAPP)
+
+        public Request(IDictionary<string, object> arbitraryKeyValuePairs, HttpRequest httpRequest = null)
+            : base(arbitraryKeyValuePairs)
+        {
+            if (httpRequest != null)
+            {
+                this.SnapProperties(httpRequest);
+            }
+        }
+
+        private void SnapProperties(HttpRequest httpRequest)
+        {
+            Assumption.AssertNotNull(httpRequest, nameof(httpRequest));
+
+            this.Url = httpRequest.Host.Value + httpRequest.Path;
+            this.QueryString = httpRequest.QueryString.Value;
+            //this.Params = null;
+
+            this.Headers = new Dictionary<string, string>(httpRequest.Headers.Count());
+            foreach (var header in httpRequest.Headers)
+            {
+                this.Headers.Add(header.Key, StringUtility.Combine(header.Value, ", "));
+            }
+
+            this.Method = httpRequest.Method;
+        }
+#endif
+
         /// <summary>
-        /// Initializes a new instance of the <see cref="Request"/> class.
+        /// Initializes a new instance of the <see cref="Request" /> class.
         /// </summary>
         /// <param name="arbitraryKeyValuePairs">The arbitrary key value pairs.</param>
-        public Request(IDictionary<string, object> arbitraryKeyValuePairs) 
+        /// <param name="httpRequest">The HTTP request.</param>
+        public Request(IDictionary<string, object> arbitraryKeyValuePairs, HttpRequestMessage httpRequest = null) 
             : base(arbitraryKeyValuePairs)
         {
+            if (httpRequest != null)
+            {
+                this.SnapProperties(httpRequest);
+            }
+        }
+
+        private void SnapProperties(HttpRequestMessage httpRequest)
+        {
+            Assumption.AssertNotNull(httpRequest, nameof(httpRequest));
+
+            this.Url = httpRequest.RequestUri.AbsoluteUri;
+            this.QueryString = httpRequest.RequestUri.Query;
+            this.Params = null;
+
+            this.Headers = new Dictionary<string, string>(httpRequest.Headers.Count());
+            foreach (var header in httpRequest.Headers)
+            {
+                this.Headers.Add(header.Key, StringUtility.Combine(header.Value, ", "));
+            }
+
+            this.Method = httpRequest.Method.Method;
+            switch(this.Method.ToUpper())
+            {
+                case "POST":
+                    var task = httpRequest.Content.ReadAsStringAsync();
+                    task.Wait();
+                    this.PostBody = task.Result;
+                    this.PostParams = null;
```

Comment:
Also here?

Reply:
these are leftovers of placeholders to be eventually populated with valid values (when available).
cleaning them up...